Add yaml dependency and use it for parsing YAML

zkry opened the pull request:
This PR adds the yaml dependency located at https://github.com/zkry/yaml.el/ (https://melpa.org/#/yaml). Using a specialized YAML parser for this simplifies the code and makes it more resilient to syntactical variations that may occur. This dependency may also ease the extraction of different metadata later on.

tarsius commented:
Thanks!

Before I merge this, could you please address the following issue.

```
lib/yaml/yaml.el:636:1: Warning: defconst `yaml--terminal-rules' docstring
    wider than 80 characters
```

By the way, I am unable to parse that sentence anyway:

> List of rules that indicate that the parse tree should end at and children ignored.

The next doc-string (of `yaml--walk-events`) is also too long. Not sure why the byte-compiler doesn't complain about that one, but you should shorten it anyway.

Speaking of long lines...
I would like to encourage you to stay below 80 characters. If the occasional line is up to 95 characters long, then that is also still okay. But that line (2106) that is 197 characters lines long without there being any benefit whatsoever to that, that's ... not so nice. And 150, 120 and even "just" 100 are also rather problematic.

I strongly urge you to stay around 80 with the occasional 95 overflow exception.

To ease that you might want to add `(declare (indent 2))` to `yaml--rep2`, and declarations for other functions similar.

zkry commented:
Thanks for the suggestions! I updated the yaml package so that lines don't exceed 80 characters. Yeah, that docstring was poorly written. I updated that as well (those were the grammar rules that were recursive on text, going really deep causing an error sometimes).

tarsius commented:
Great. Instantly looks better. :grin:

I intend to merge this with the above fixup changes.
